runtime/LibcWrappers.cpp

```diff
@@ -246,6 +246,26 @@ size_t SYM(fread)(void *ptr, size_t size, size_t nmemb, FILE *stream) {
   return result;
 }
 
+char *SYM(fgets)(char *str, int n, FILE *stream) {
+  tryAlternative(str, _sym_get_parameter_expression(0), SYM(fgets));
+  tryAlternative(n, _sym_get_parameter_expression(1), SYM(fgets));
+
+  auto result = fgets(str, n, stream);
+  _sym_set_return_expression(nullptr);
+
+  if (fileno(stream) == inputFileDescriptor) {
+    // Reading symbolic input.
+    ReadWriteShadow shadow(str, sizeof(char) * n);
```

Comment:
According to the man page, `fgets` reads at most `n-1` characters, and it also stops at EOF or after a newline. I believe that using `n` here unconditionally may lead to a wrong increment of `inputOffset` below (e.g., if `fgets` is called with a large buffer but ends up reading just a couple of characters before it encounters the end of the line). Shouldn't we use `strlen(str)` instead? The same concern applies to the case below where we erase the shadow.

Reply:
I applied the suggested changes :)